src/Analyser/TypeSpecifier.php

```diff
@@ -267,6 +267,47 @@ public function specifyTypesInCondition(
 						);
 					}
 				}
+
+				if (
+					!$context->null()
+					&& $exprNode instanceof FuncCall
+					&& $exprNode->name instanceof Name
+					&& strtolower((string) $exprNode->name) === 'get_parent_class'
+					&& isset($exprNode->getArgs()[0])
+					&& $constantType instanceof ConstantStringType
+				) {
+					$argType = $scope->getType($exprNode->getArgs()[0]->value);
+					$objectType = new ObjectType($constantType->getValue());
```

Comment:
the reported phpstan error is interessting..

![grafik](https://user-images.githubusercontent.com/120441/171575508-e586c48b-d3d4-405c-aef5-b3e4ff9bc659.png)

shouldn't phpstan know, that `$constantType is a ConstantStringType` and therefore know that `getValue` can only return a string?

Reply:
It should and it works in other places too. My guess is that you broke this functionality in this PR, but it's hard to say just by looking. You need to debug it.

Reply:
the issue I am experiencing here seems not be triggered by my changes. I think https://github.com/phpstan/phpstan-src/pull/1396 proves that the problem also happens on the latest master without further modifications